backend/src/main/kotlin/com/tngtech/apicenter/backend/connector/rest/controller/VersionController.kt

```diff
@@ -25,27 +23,28 @@ class VersionController constructor(private val versionHandler: VersionHandler,
             headers =  ["Accept=" + MediaType.APPLICATION_JSON_VALUE,
                         "Accept=" + MEDIA_TYPE_YAML],
             method =   [RequestMethod.GET])
-    fun findVersion(@PathVariable specificationId: UUID,
+    fun findVersion(@PathVariable specificationId: String,
                     @PathVariable version: String,
                     @RequestHeader(value = "Accept",
                                    defaultValue = MediaType.APPLICATION_JSON_VALUE) accept: String = MediaType.APPLICATION_JSON_VALUE): VersionDto {
         // i.e. The integration test and unit test require the default specified in two different ways
-        val foundVersion = versionHandler.findOne(specificationId, version)
-        if (foundVersion == null) {
-            throw SpecificationNotFoundException(specificationId, version)
-        } else if (accept == MEDIA_TYPE_YAML) {
+        val foundVersion = versionHandler.findOne(specificationId, version) ?: throw SpecificationNotFoundException(specificationId, version)
+
+        val convertedVersion = if (accept == MEDIA_TYPE_YAML) {
             logger.info("Specification $specificationId version $version requested as YAML")
             val jsonNodeTree = ObjectMapper().readTree(foundVersion.content)
             val jsonAsYaml = YAMLMapper().writeValueAsString(jsonNodeTree)
-            return versionDtoMapper.fromDomain(Version(jsonAsYaml, foundVersion.metadata))
+            Version(jsonAsYaml, foundVersion.metadata)
```

Comment:
I'm not a fan of the short-hand `val blah = if () { [a lot of stuff happening] suddenly an object appears }`. Again, a good alternative is to extract another small method.